archiver/archiver.go

```diff
@@ -162,14 +163,17 @@ func (a *archiver) Start() error {
 		// pinger loop
 		for {
 			if a.config.enabled() {
+				start := time.Now()
 				sources, err := a.jobsDB.GetDistinctParameterValues(ctx, "source_id")
+				a.stats.NewStat("arc_active_partitions_time", stats.TimerType).Since(start)
```

Comment:
We can prepare this stat once as part of the setup? The same goes with the below one as well.

Reply:
Caching previously-created stats is already taken care of for us by the stats package, I prefer not having yet-another-field